driver/src/main/java/oracle/nosql/driver/httpclient/HttpClient.java

```diff
@@ -189,33 +189,40 @@ public HttpClient(String host,
         this.maxChunkSize = (maxChunkSize == 0 ?
                               DEFAULT_MAX_CHUNK_SIZE : maxChunkSize);
 
-        workerGroup = new NioEventLoopGroup(numThreads);
+        int cores = Runtime.getRuntime().availableProcessors();
 
+        if (numThreads == 0) {
+            numThreads = cores*2;
+        }
+        if (connectionPoolSize == 0) {
+            connectionPoolSize = cores*2;
+        }
+        workerGroup = new NioEventLoopGroup(12); //numThreads);
```

Comment:
12?

Reply:
oops, debugging code left in, should be numThreads. Thanks.